lib/core/src/Cardano/Wallet/DB.hs

```diff
@@ -329,6 +331,8 @@ newtype ErrNoSuchWallet
     = ErrNoSuchWallet WalletId -- Wallet is gone or doesn't exist yet
     deriving (Eq, Show)
 
+instance Exception ErrNoSuchWallet
```

Comment:
The possibility of both checked and unchecked exceptions - this is something we want to avoid.